Strain/StrainVisualization.cpp

```diff
@@ -0,0 +1,117 @@
+#include "StrainVisualization.h"
+
+#include <iostream>
+
+
+void StrainVisualization::draw(sf::RenderTarget& target, sf::RenderStates states) const
+{
+	target.draw(_coordSystem);
+	target.draw(_torqueX);
+	target.draw(_torqueY);
+	target.draw(_torqueZ);
+	target.draw(_forceX);
+	target.draw(_forceY);
+	target.draw(_forceZ);
+}
+
+StrainVisualization::StrainVisualization(const int maxX, const int maxY, const LPCTSTR port)
+	: StrainGauge(port),
+	  _limitX(maxX),
+	  _limitY(maxY),
+	  _calibrate({{0, 0, 0, 0, 0, 0}})
+{
+	_coordSystem = sf::VertexArray(sf::LineStrip, 4);
+	_coordSystem.append(sf::Vertex(sf::Vector2f(1, 0), sf::Color::Red));
+	_coordSystem.append(sf::Vertex(sf::Vector2f(1, 2 * _limitY), sf::Color::Red));
+	_coordSystem.append(sf::Vertex(sf::Vector2f(1, _limitY), sf::Color::Red));
+	_coordSystem.append(sf::Vertex(sf::Vector2f(_limitX, _limitY), sf::Color::Red));
+
+	_forceX = sf::VertexArray(sf::LineStrip, _limitX);
+	_forceY = sf::VertexArray(sf::LineStrip, _limitX);
+	_forceZ = sf::VertexArray(sf::LineStrip, _limitX);
+	_torqueX = sf::VertexArray(sf::LineStrip, _limitX);
+	_torqueY = sf::VertexArray(sf::LineStrip, _limitX);
+	_torqueZ = sf::VertexArray(sf::LineStrip, _limitX);
+
+	_y1 = std::vector<int>(_limitX);
+	_y2 = std::vector<int>(_limitX);
+	_y3 = std::vector<int>(_limitX);
+	_m1 = std::vector<int>(_limitX);
+	_m2 = std::vector<int>(_limitX);
+	_m3 = std::vector<int>(_limitX);
+}
+
+void StrainVisualization::strainCalibrating()
+{
+	const int numberOfReadings = 50;
```

Comment:
**(must be changed)**: change to `constexpr`.

Reply:
Done